### DISCUSSION TITLE
Use SyncDirection.UploadOnly to insert new row

### DISCUSSION THREAD

**Discussion by josueascencio** [Mimetis/Dotmim.Sync#543] [Q&A]
Dear, thank you very much for sharing knowledge.
I am doing a project in which I only need to extract the information from the client without it being deleted or updated on the server. I am using SyncDirection.UploadOnly, but it updates the row on the server, I need it to insert a new one. What could you recommend me?

Thank you

**Comment by Mimetis** [comment]
This is the expected behavior.
When you are **adding** / **editing** or **deleting** a row on your client, this row will be **upload** to the server and **synced**.
But any row that is actually **added** / **edited** or **deleted** from the server will never be **downloaded** to any client.

I don't understand what you want to accomplish here ?
Can you explain more in details your scenario ?

**Reply by josueascencio** [reply]
Thanks for your answer.

The scenario I am looking for is that clients can only upload information, but multiple clients will do the same.
For example: If client "A" sends its information to the server to the products table occupies rows 1 and 2. Client "B" also sends new information to the products table, it should use rows 3 and 4 and not upsert rows 1 and 2.

I hope I can explain it.
And even better I hope that DMS can be used in my scenario.

**Reply by Mimetis** [reply]
I suppose you are using `Identity `columns to store your rows ?
If it's the case, I guess you need to modify your schema and use `GUID` as a replacement for primary keys

**Reply by josueascencio** [reply]
That's correct, I use `Identity.`
In this case the modification should be done directly in SQL?
For DMS should it be transparent or should I modify something in my client?

**Reply by Mimetis** [reply]
Have a look here : https://github.com/Mimetis/Dotmim.Sync/discussions/442#discussioncomment-361975

**Reply by josueascencio** [reply]
I'll take a look now. Thank you for your valuable time.
